cat/opencl.cat

```diff
@@ -0,0 +1,128 @@
+OpenCL
+(* OpenCL Memory Model *)
+include "basic.cat"
+
+(*
+* This model is based on:
+* https://multicore.doc.ic.ac.uk/overhauling/opencl_base.cat
+* https://multicore.doc.ic.ac.uk/overhauling/opencl_scopedsc.cat
+*)
+
+// Base relations:
+// int: same work-item
+// swg: same work-group
+// sdv: same device
+// syncbar: same barrier id
+
+// Tags:
+// WI: work-item scope
+// WG: work-group scope
+// DV: device scope
+// ALL: all-svm-devices scope
+// GLOBAL: global memory
+// LOCAL: local memory
+// EF: entry fence of barrier
+// XF: exit fence of barrier
+
+// dynamic_tag relates events to itself that access an address whose init event is marked X or Fence tagged with X
+let dynamic_tag(X) = [range([IW & X]; loc)] | [X & F]
+
+let mo = dynamic_tag(~NAL); co; dynamic_tag(~NAL)
+let sb = po
+let rb = (rf^-1; mo) \ id
```

Comment:
You need to add the thing to deal with memory not having initial events and you don't need the \id ... this is for handling RMW as a single event.

In short, replace this by the definition of `fr` in c11.cat

Reply:
Updated